### PR TITLE
[WIP] Faster rpm build

### PR DESCRIPTION
**Pull request by jrafanie**
Here are some very WIP small and larger changes to try to speed up the rpm build.

My plan is to extract things we like and get them in.
* disable pythondeps - wastes 45 minutes locally... not sure if we need it, and definitely shouldn't be changing all that often.  We can possibly run this once per major change.
* final commit is just debug logging output

EDIT: 
Extracted to https://github.com/ManageIQ/manageiq-rpm_build/pull/437
* .build-id artifacts - To extract, pretty minor... tens of files are eliminated from the tarballs
* use built-in vcs to catch .git, .gitignore and friends instead of hardcoded list - seems safe to extract and it's super minor
* faster brp-strip... was 33 minutes... now around 2.  Uses new version of the file.

### CODE REVIEW

**Comment by jrafanie** [comment]
> Another option might be to just install rpmbuild 4.19 directly to avoid the curl and os macro changes. Not sure how hard that is or not

My assumption is we'd need to add a new dnf repo to install it as it will likely pull in lots of other dependencies. We're using the latest rpm/rpmbuild available in our ubi8 repo.

File: lib/manageiq/rpm_build/generate_tar_files.rb [hunk 1]
```diff
@@ -22,14 +33,14 @@ def create_gemset_tarball
         plugin_index.write(plugin_index.read.gsub(BUILD_DIR.join("manageiq").to_s, '/var/www/miq/vmdb'))
 
         name = "gemset"
-        shell_cmd("tar -C #{BUILD_DIR} -zcf #{tar_full_path(name)} -X #{exclude_file(name)} #{tar_basename(name)}")
+        shell_cmd("tar -C #{BUILD_DIR} -zcf #{tar_full_path(name)} --exclude-vcs -X #{exclude_file(name)} #{tar_basename(name)}")
```
[the remark this answers is not in the document]
Comment (jrafanie) [reply lib/manageiq/rpm_build/generate_tar_files.rb:36]: Good point.  I really want to see if I can run rpmdiff before and after this PR.  I will not extract that until I can do that.

File: rpm_spec/brp-strip-with-ruby-js-filtering [hunk 2]
```diff
@@ -0,0 +1,50 @@
+#!/bin/sh
+# From https://github.com/rpm-software-management/rpm/blob/rpm-4.19.1-release/scripts/brp-strip
```
[the remark this answers is not in the document]
Comment (jrafanie) [reply rpm_spec/brp-strip-with-ruby-js-filtering:2]: yes, I think it's similar to pulling down the source for repmgr in https://github.com/ManageIQ/manageiq-rpm_build/blob/master/packages/repmgr13/repmgr.spec#L24.  Then, we just have patch file with the minor change to add rb/js filtering and we can apply that on top.

**Comment by Fryguy** [comment]
We also discussed a possible `rm -rf` of all python test directories.  While that might not change much since disabling pythondeps is the bigger change, it might have other benefits, such as
- Smaller RPM files
- Even less to traverse as part of brp-strip
- Removing files that are subject to license auditing (I actually get these frequently because some test libraries have copies of GPL files within them)

File: Dockerfile [hunk 3]
```diff
@@ -63,6 +63,18 @@ RUN npm install yarn -g
...
+
+RUN curl -o brp-strip https://raw.githubusercontent.com/rpm-software-management/rpm/rpm-4.19.1-release/scripts/brp-strip
+
+RUN chmod +x brp-strip
+
+RUN patch -p2 < Add-js-rb-filtering-on-top-of-4.19.1.patch && \
+  ls -l /usr/lib/rpm/brp-strip && \
+  mv brp-strip /usr/lib/rpm/brp-strip && \
+  ls -l /usr/lib/rpm/brp-strip && \
+  rm -f Add-js-rb-filtering-on-top-of-4.19.1.patch
```
Comment (Fryguy) [inline Dockerfile:76]: Nice!  A few comments here:

- The patch file should probably live in the container-assets directory, as opposed to the root.  That way you can reuse the `COPY . /build_scripts` command which already copies the container-assets directory.
- Prefer curling the file directly into /usr/lib/rpm/brp-strip, overwriting it, and then patching /usr/lib/rpm/brp-strip directly.
- No need to remove the patch file (it doesn't really get deleted from the container anyway)

File: Dockerfile [hunk 4]
```diff
@@ -63,8 +63,15 @@ RUN npm install yarn -g
...
 
+RUN curl -o /usr/lib/rpm/brp-strip https://raw.githubusercontent.com/rpm-software-management/rpm/rpm-4.19.1-release/scripts/brp-strip
+
+RUN chmod +x /usr/lib/rpm/brp-strip
+
 COPY . /build_scripts
 
+RUN cd /usr/lib/rpm/ && \
+  patch -p2 < /build_scripts/container-assets/Add-js-rb-filtering-on-top-of-4.19.1.patch
+
```
Comment (Fryguy) [inline Dockerfile:74]: You can still smash all these together further

```suggestion
COPY . /build_scripts

RUN curl -o /usr/lib/rpm/brp-strip https://raw.githubusercontent.com/rpm-software-management/rpm/rpm-4.19.1-release/scripts/brp-strip && \
  chmod +x /usr/lib/rpm/brp-strip && \
  cd /usr/lib/rpm/ && \
  patch -p2 < /build_scripts/container-assets/Add-js-rb-filtering-on-top-of-4.19.1.patch
```

**Comment by Fryguy** [comment]
Merged #437, so this needs a rebase.

File: rpm_spec/brp-strip-with-ruby-js-filtering (hunk 2 above)
[the remark this answers is not in the document]
Comment (kbrock) [reply rpm_spec/brp-strip-with-ruby-js-filtering]: Does this bring in the exclude `js`/`rb` changes we got merged into the tool?
I remember getting it merged, but can't remember what version of the rpm tool it was.

**Comment by jrafanie** [comment]
#437 has the commits we decided to extract already.  The remaining work will not be done at this time.  I can reopen if it's ever wanted.

**Pull request closed without merging**